lib/booking_manager.rb

```diff
@@ -0,0 +1,125 @@
+require_relative 'room'
+require_relative 'single_res'
+require_relative 'block_res'
+require_relative 'date_range'
+
+module Hotel
+  class BookingManager
+    attr_reader :rooms, :reservations, :blocks
+    
+    def initialize(total_room_count)
+      @rooms = Hotel::Room.all_rooms(total_room_count)
+      @reservations = []
+      @blocks = []
+    end
+    
+    def list_all_rooms
+      room_list = @rooms.map do |room|
+        "Room #{room.number}"
+      end.join("\n")
+      
+      return room_list
+    end
+    
+    def find_reservations_by_date(date)
+      found_singles = reservations.select do |reservation|
+        reservation.date_range.date_included?(date)
+      end
+
+      found_blocks = blocks.select do |block|
+        block.date_range.date_included?(date)
+      end
+
+      found_reservations = found_singles + found_blocks
+      
+      if found_reservations.length == 0
+        nil
+      else
+        return found_reservations
+      end 
+    end
+
+    def rooms_available(start_date, end_date)
+      reserved_single = reservations.select do |reservation|
+        reservation.date_range.overlaps?(start_date, end_date)
+      end.map! {|reservation| reservation.room}
+
+      reserved_blocks = blocks.select do |block|
+        block.date_range.overlaps?(start_date, end_date)
+      end.map! {|block| block.unreserved_rooms}
+
+      if reserved_single.length > 0 && reserved_blocks.length > 0
+        all_reserved = reserved_single + reserved_blocks.flatten!
+      elsif reserved_single.length > 0 && reserved_blocks.length == 0
+        all_reserved = reserved_single
+      elsif reserved_single.length == 0 && reserved_blocks.length > 0
+        all_reserved = reserved_blocks.flatten!
+      else
+        all_reserved = []
+      end
+      
+      available_rooms = rooms - all_reserved
+
+      return available_rooms
+    end
+
+    def find_rooms(available_rooms, total_rooms = 1)
+      if total_rooms == 1
+        return available_rooms[0]
+      else
+        rooms_array = []
+        total_rooms.times do
+          rooms_array << available_rooms.shift
+        end
+        return rooms_array
+      end
+    end
```

Comment:
Hm, I'm not sure what this method does besides take the first X rooms from `available_rooms`? Also, I think it's not explicitly tested ;P (Not high priority feedback)